diagnostic.js

```diff
@@ -27,28 +27,69 @@
 const mongoose = require('mongoose');
 mongoose.Promise = global.Promise;
 mongoose.connect('mongodb://localhost/mongoose-crud');
+const Movement = require('models/movement.js'); // model
 
 /// ADD YOUR CODE BELOW
 
-const create = (name, description, startYear, endYear) => {};
+const create = (name, description, startYear, endYear) => {
+  Movement.create({
+    name: name,
+    description: description,
+    startYear:startYear,
+    endYear:endYear,
+  }).then(function(movement) {
+    console.log(place);
+  }).catch(function(error) {
+    console.error(error);
+  }).then(done);
+
+};
 // Success -> Print new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  Movement.find({})
+   .then(function(movements){
+     movements.forEach(function(movement) {
+       console.log(movement.toJSON());
+     });
+   }).catch(console.error)
+
+};
 // Success -> Print all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+const show = (id) => {
+  Movement.findById(id).then(function(movement) {
+    console.log(movement.toObject());
+  }).catch(console.error).then(done);
```

Comment:
No `done`. Also, you're missing an edge case - what happens if the ID doesn't match any documents. Not finding a record is not the same as an error; in that case, `findByID` spits out a `null` instead of a record. If that happens, you should NOT try to call `.toJSON` on it because that _will_ cause an error. Instead, we should test whether or not `movement` is not null, and if it is null, we should print out the text "NOT FOUND" and exit.